exporters/jaeger/uploader.go

```diff
@@ -112,8 +114,18 @@ func WithAgentPort(port string) AgentEndpointOption {
 	})
 }
 
+var emptyLogger = logr.Logger{}
+
 // WithLogger sets a logger to be used by agent client.
 func WithLogger(logger *log.Logger) AgentEndpointOption {
+	return agentEndpointOptionFunc(func(o agentEndpointConfig) agentEndpointConfig {
+		o.Logger = stdr.New(logger)
+		return o
+	})
+}
+
+// WithLogr sets a logr.Logger to be used by agent client.
+func WithLogr(logger logr.Logger) AgentEndpointOption {
```

Comment:
```suggestion
func WithLogger(logger *log.Logger) AgentEndpointOption {
	return WithLogr(stdr.New(logger))
}

// WithLogr sets a logr.Logger to be used by agent client.
func WithLogr(logger logr.Logger) AgentEndpointOption {
```

I think I would implement one in terms of the other.

Reply:
Yes, it's better this way, updated